waku/v2/protocol/filter/filter_push_test.go

```diff
@@ -141,3 +144,84 @@ func (s *FilterTestSuite) TestLargePayloadsUTF8() {
 	s.Require().NoError(err)
 
 }
+
+func (s *FilterTestSuite) TestTimestampInFuture() {
+
+	// Subscribe
+	s.subDetails = s.subscribe(s.testTopic, s.testContentTopic, s.fullNodeHost.ID())
+
+	// Set time in the future
+	timeDelta, _ := time.ParseDuration("200h")
+	futureTime := proto.Int64(time.Now().UnixNano() + timeDelta.Nanoseconds())
+	message := tests.CreateWakuMessage(s.testContentTopic, futureTime, "test_payload")
+
+	// Should get accepted
+	_, err := s.relayNode.Publish(s.ctx, message, relay.WithPubSubTopic(s.testTopic))
+	s.Require().NoError(err)
+
+	// Should be received
+	s.waitForMsg(nil, s.subDetails[0].C)
+
+	_, err = s.lightNode.UnsubscribeAll(s.ctx)
+	s.Require().NoError(err)
+
+}
+
+func (s *FilterTestSuite) TestZeroTimestamp() {
+
+	// Subscribe
+	s.subDetails = s.subscribe(s.testTopic, s.testContentTopic, s.fullNodeHost.ID())
+
+	message := tests.CreateWakuMessage(s.testContentTopic, new(int64), "test_payload")
+
+	// Should get accepted
+	_, err := s.relayNode.Publish(s.ctx, message, relay.WithPubSubTopic(s.testTopic))
+	s.Require().NoError(err)
+
+	// Should be received
+	s.waitForMsg(nil, s.subDetails[0].C)
+
+	_, err = s.lightNode.UnsubscribeAll(s.ctx)
+	s.Require().NoError(err)
+
+}
+
+func (s *FilterTestSuite) TestNegativeTimestamp() {
+
+	// Subscribe
+	s.subDetails = s.subscribe(s.testTopic, s.testContentTopic, s.fullNodeHost.ID())
+
+	negTimestamp := int64(-100)
+	message := tests.CreateWakuMessage(s.testContentTopic, &negTimestamp, "test_payload")
+
+	// Should get accepted
+	_, err := s.relayNode.Publish(s.ctx, message, relay.WithPubSubTopic(s.testTopic))
```

Comment:
Again, trying to understand the objective of this test. 

Is it just to verify that message can be pushed wit negative timestamp?

Reply:
Yes, not much of the value, testing grey zone. Removed for this PR.